trunk-recorder/main.cc

```diff
@@ -1026,7 +1042,11 @@ void handle_message(std::vector<TrunkMessage> messages, System *sys) {
     switch (message.message_type) {
     case GRANT:
     case UPDATE:
-      handle_call(message, sys);
+      if ((message.talkgroup != 0) && (message.freq != 0)) {
+        handle_call(message, sys);
+      } else {
+        BOOST_LOG_TRIVIAL(trace) << "Skipping call: Freq " << message.freq << " TG " << message.talkgroup << " Src " << message.source;
```

Comment:
Why would talkgroup or the Freq be 0? If this is a bad message, lets handle it further up the processing stream, in the parser.

Reply:
I've consistently (as in continuously) seen messages with them. I want to say they even show up in UniTrunker. I'm thinking it's not due to poor signal. I could send you some raw signal if you want to see it.

Reply:
Gotcha - This could be another processing error. I have never see this in my system. The error checking as part of the decoding should stop malformed packets because of low signal strength but maybe some get through. Can you turn on Tracing level debug so we can see exactly which message parsing function is causing this? This error check should be integrated directly into the parsing.

Could you also a Radio Reference link for this system? I am curious if there could be some special manufacturer messages in the control channel.

Let's leave this out of the PR and handle it separately.

Reply:
I don't see the problem with it, but I can definitely leave it out. As I said, I think it's not a result of malformed packets.

Your mention of trace debug level reminded me of how trunk-recorder can show the raw data bits (which would include packet check sums). I should be able to ferret out the ones in question and copy them here. [This is the system I primarily monitor](https://www.radioreference.com/apps/db/?sid=7202); I have at least 2 other P25 systems I can check too.